cmake/kokkos_arch.cmake

```diff
@@ -67,9 +67,16 @@ KOKKOS_ARCH_OPTION(ZEN3            HOST "AMD Zen3 architecture")
 KOKKOS_ARCH_OPTION(VEGA900         GPU  "AMD GPU MI25 GFX900")
 KOKKOS_ARCH_OPTION(VEGA906         GPU  "AMD GPU MI50/MI60 GFX906")
 KOKKOS_ARCH_OPTION(VEGA908         GPU  "AMD GPU MI100 GFX908")
-KOKKOS_ARCH_OPTION(INTEL_GEN       GPU  "Intel GPUs Gen9+")
-
+KOKKOS_ARCH_OPTION(INTEL_GEN       GPU  "Intel GPUs Gen9+ (deprecated)")
 
+KOKKOS_OPTION(ARCH_INTEL_GPU "" STRING "Intel GPU architecture used for ahead-of-time compilation")
+IF(KOKKOS_ARCH_INTEL_GEN)
+  # deprecated
+  SET(KOKKOS_ARCH_INTEL_GPU skl CACHE STRING "Intel GPU architecture used for ahead-of-time compilation" FORCE)
```

Comment:
Should we emit a warning?

Reply:
Possibly, I'll wait for the outcome of the discussion before continuing here, though.